### ISSUE TITLE
Yolov8 detect train class imbalanced problem

### ISSUE THREAD

**Issue by duymanh-111** [ultralytics/ultralytics#5738]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Dear Ultralytics Team!
I have my custom dataset to classify 30 classes.
But the classes's count are very different.
For example one class1's count is 200, class2's count is 30000.
I trained with this imbalanced dataset, the box detection is good but class detection is very bad.
Is there the parameter in yolov8 training params to solve the class imbalanced problem without manual data augment.
Thanks.

**Comment by glenn-jocher** [comment]
@duymanh-111 hello,

Thank you for reaching out to us on this issue! In situations with imbalance in class distributions, it is common to observe the performance of a model to be suboptimal, especially for the classes with fewer representations in the dataset.

Within the Ultralytics YOLOv8 implementation, we have considered this point. You may adjust the 'cls' loss weight in the loss configuration of your training script. Increasing this value makes the model pay more attention to classification accuracy, which could help to alleviate the imbalance problem.

Though the 'cls' loss weight adjustment can help, it might not fully solve the imbalance problem for extremely skewed distributions. Hence, for robust performance and optimal results, generating additional representations for the under-represented classes using data augmentation techniques, although manual, is still highly recommended.

On another note, the model's predictions and the class score output can be modulated using the confidence thresholding and non-maximum suppression (NMS) parameters during inference, which can also contribute to improving overall class detection performance.

I hope this information assists you in improving the performance of your model, and we'd like to hear how you get on. 

Best,
Glenn Jocher

**Comment by Y-T-G** [comment]
Duplicate of https://github.com/ultralytics/ultralytics/issues/21082
